packages/ledger-icrc/src/utils/ledger.utils.ts

```diff
@@ -89,3 +95,60 @@ export const decodeIcrcAccount = (accountString: string): IcrcAccount => {
 
   return account;
 };
+
+/**
+ * Maps the token metadata information from a ledger response into a structured record.
+ *
+ * This utility processes an array of metadata key-value pairs provided by the ledger
+ * and extracts specific fields, such as symbol, name, fee, decimals, and logo. It then
+ * constructs a `IcrcTokenMetadata` record. If any required fields are missing,
+ * the function returns `undefined`.
+ *
+ * @param {IcrcTokenMetadataResponse} response - An array of key-value pairs representing token metadata.
+ *
+ * @returns {IcrcTokenMetadata | undefined} - A structured metadata record or `undefined` if required fields are missing.
+ */
+export const mapTokenMetadata = (
+  response: IcrcTokenMetadataResponse,
+): IcrcTokenMetadata | undefined => {
+  const nullishToken = response.reduce<Partial<IcrcTokenMetadata>>(
+    (acc, [key, value]) => {
+      switch (key) {
+        case IcrcMetadataResponseEntries.SYMBOL:
+          acc = { ...acc, ...("Text" in value && { symbol: value.Text }) };
+          break;
+        case IcrcMetadataResponseEntries.NAME:
+          acc = { ...acc, ...("Text" in value && { name: value.Text }) };
+          break;
+        case IcrcMetadataResponseEntries.FEE:
+          acc = { ...acc, ...("Nat" in value && { fee: value.Nat }) };
```

Comment:
just to double-check, no need to do `Number(value.Nat)` here?

Reply:
Correct, we keep fee in bigint. Note this code was copied from OISY: